pkg/gui/keybindings.go

```diff
@@ -28,178 +32,192 @@ type Binding struct {
+	{gocui.KeyCtrlU, gocui.ModCtrl}:      "ctrl+u",
+	{gocui.KeyCtrlV, gocui.ModCtrl}:      "ctrl+v",
+	{gocui.KeyCtrlW, gocui.ModCtrl}:      "ctrl+w",
+	{gocui.KeyCtrlX, gocui.ModCtrl}:      "ctrl+x",
+	{gocui.KeyCtrlY, gocui.ModCtrl}:      "ctrl+y",
+	{gocui.KeyCtrlZ, gocui.ModCtrl}:      "ctrl+z",
+	{gocui.KeyCtrl4, gocui.ModCtrl}:      "ctrl+4", // ctrl+\
+	{gocui.KeyCtrl5, gocui.ModCtrl}:      "ctrl+5", // ctrl+]
+	{gocui.KeyCtrl6, gocui.ModCtrl}:      "ctrl+6",
+	{gocui.KeyCtrl8, gocui.ModCtrl}:      "ctrl+8",
 }
 
-var keymap = map[string]interface{}{
-	"<c-a>":       gocui.KeyCtrlA,
-	"<c-b>":       gocui.KeyCtrlB,
-	"<c-c>":       gocui.KeyCtrlC,
-	"<c-d>":       gocui.KeyCtrlD,
-	"<c-e>":       gocui.KeyCtrlE,
-	"<c-f>":       gocui.KeyCtrlF,
-	"<c-g>":       gocui.KeyCtrlG,
-	"<c-h>":       gocui.KeyCtrlH,
-	"<c-i>":       gocui.KeyCtrlI,
-	"<c-j>":       gocui.KeyCtrlJ,
-	"<c-k>":       gocui.KeyCtrlK,
-	"<c-l>":       gocui.KeyCtrlL,
-	"<c-m>":       gocui.KeyCtrlM,
-	"<c-n>":       gocui.KeyCtrlN,
-	"<c-o>":       gocui.KeyCtrlO,
-	"<c-p>":       gocui.KeyCtrlP,
-	"<c-q>":       gocui.KeyCtrlQ,
-	"<c-r>":       gocui.KeyCtrlR,
-	"<c-s>":       gocui.KeyCtrlS,
-	"<c-t>":       gocui.KeyCtrlT,
-	"<c-u>":       gocui.KeyCtrlU,
-	"<c-v>":       gocui.KeyCtrlV,
-	"<c-w>":       gocui.KeyCtrlW,
-	"<c-x>":       gocui.KeyCtrlX,
-	"<c-y>":       gocui.KeyCtrlY,
-	"<c-z>":       gocui.KeyCtrlZ,
-	"<c-~>":       gocui.KeyCtrlTilde,
-	"<c-2>":       gocui.KeyCtrl2,
-	"<c-3>":       gocui.KeyCtrl3,
-	"<c-4>":       gocui.KeyCtrl4,
-	"<c-5>":       gocui.KeyCtrl5,
-	"<c-6>":       gocui.KeyCtrl6,
-	"<c-7>":       gocui.KeyCtrl7,
-	"<c-8>":       gocui.KeyCtrl8,
-	"<c-space>":   gocui.KeyCtrlSpace,
-	"<c-\\>":      gocui.KeyCtrlBackslash,
-	"<c-[>":       gocui.KeyCtrlLsqBracket,
-	"<c-]>":       gocui.KeyCtrlRsqBracket,
-	"<c-/>":       gocui.KeyCtrlSlash,
-	"<c-_>":       gocui.KeyCtrlUnderscore,
-	"<backspace>": gocui.KeyBackspace,
-	"<tab>":       gocui.KeyTab,
-	"<backtab>":   gocui.KeyBacktab,
-	"<enter>":     gocui.KeyEnter,
-	"<a-enter>":   gocui.KeyAltEnter,
-	"<esc>":       gocui.KeyEsc,
-	"<space>":     gocui.KeySpace,
-	"<f1>":        gocui.KeyF1,
-	"<f2>":        gocui.KeyF2,
-	"<f3>":        gocui.KeyF3,
-	"<f4>":        gocui.KeyF4,
-	"<f5>":        gocui.KeyF5,
-	"<f6>":        gocui.KeyF6,
-	"<f7>":        gocui.KeyF7,
-	"<f8>":        gocui.KeyF8,
-	"<f9>":        gocui.KeyF9,
-	"<f10>":       gocui.KeyF10,
-	"<f11>":       gocui.KeyF11,
-	"<f12>":       gocui.KeyF12,
-	"<insert>":    gocui.KeyInsert,
-	"<delete>":    gocui.KeyDelete,
-	"<home>":      gocui.KeyHome,
-	"<end>":       gocui.KeyEnd,
-	"<pgup>":      gocui.KeyPgup,
-	"<pgdown>":    gocui.KeyPgdn,
-	"<up>":        gocui.KeyArrowUp,
-	"<down>":      gocui.KeyArrowDown,
-	"<left>":      gocui.KeyArrowLeft,
-	"<right>":     gocui.KeyArrowRight,
+var keymap = map[string]KeyMod{
+	"<c-a>":       {gocui.KeyCtrlA, gocui.ModCtrl},
```

Comment:
should this be gocui,KeyA? or just 'a'?

Reply:
There is no gocui.KeyA and I think there is a good reason for that. My original plan was that I'll be able to write:
"<c-a>":       {'a', gocui.ModCtrl},
Unfortunately it doesn't work because tcell itself doesn't treat <c-a> as ModCtrl + 'a'. It has a separate KeyRune for it: KeyCtrlA, and at least to my knowledge there is no easy way to know it contains an 'a' (tcell won't return 'a' for tev.Key()). We would need to decode it somewhere (tcell_driver.go probably) back to ModCtrl + 'a' if we'd really want to do it this way. I'm not sure why tcell does it this way. 

This is what tcell docs say, regarding CtrlA:
` KeyRune is special however, in that it is a place holder key indicating that a printable character was sent. The actual value of the rune will be transported in the Rune of the associated EventKey. `

I see two ways forward:
1/ Decode CtrlA and the likes back to ModCtrl + 'a'. There might be a good reason why tcell does it this way so I'm a bit afraid there are consequences of this design decision we are not aware.
2/ Let these special KeyRunes flow through gocui just the way we do it now and use them in the keymap. Maybe we could remove the modifier keys where it makes sense. So for example:
	"<c-a>":       {gocui.KeyCtrlA, gocui.ModNone},
but
       	"<c-right>":       {gocui.KeyArrowRight, gocui.ModCtrl},

I'm more than happy to try anything else if you have other ideas.

Reply:
Sounds like a doozy. I'm happy to leave it as-is in that case